### PR TITLE
stm32f2: add stm32f215xg soc

### PR DESCRIPTION
**Pull request by ibisz**
The commit is based on PR #7450 'porting stm32f2'

Used and tested with threads, hello_world samples
on out of tree custom board.

Signed-off-by: Istvan Bisz <istvan.bisz@t-online.hu>
Signed-off-by: qianfan Zhao <qianfanguijin@163.com>

### CODE REVIEW

File: arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.stm32f215xx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Kconfig - ST Microelectronics stm32f215xx MCU
+#
+# Copyright (c) 2018 qianfan Zhao <qianfanguijin@163.com>
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.stm32f215xx]: Provide your own copyright.
Even if this is copy/paste, you're the one doing the test and upstream effort.

File: arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.stm32f215xx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+if SOC_STM32F215XG
+
+config SOC
+	string
+	default "STM32F215xx"
+
+config NUM_IRQS
+	int
+	default 80
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f2/Kconfig.defconfig.stm32f215xx]: 81

File: dts/arm/st/stm32f215.dtsi [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
```
Comment (erwango) [inline dts/arm/st/stm32f215.dtsi]: Use your copyright here also

**Review by erwango — APPROVED** [review]

**Review by erwango — DISMISSED** [review]
I've approved by error.
Some  comments.

One outstanding 'issue' is that there is no board associated as tested on custom board.
@galak, we've discussed this a bit earlier. What do you expect in this case

**Comment by ibisz** [comment]
@galak @erwango Could you please merge this PR.
A new STM32F2 extra tree SOC definition isn’t possible without the hole DT structure moving to the extra tree board folder. These changes make practically impossible a tree compatible new PR definition.
The new approach of the extra tree SOCs mentioned in Out of tree BSP #9172 will solve this issue in the future, but we need now a practical solution.

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Comment by ibisz** [comment]
The PR-s:
#9776 Move soc code under ZEPHYR_PROJECT/soc 
#9980 dts/st: enables possibility to define Flash and SRAM out of tree

makes this request inutile.

**Comment by pfalcon** [comment]
@ibisz : You can just rebase your branch on the latest master and force-push to this same PR, given that it was approved already, review of the new version should be faster.

**Comment by ibisz** [comment]
@galak, the PR contains a minimal tree change to define a new SOC in stm32f2 series without an eval/discovery/nucleo board supporting it.
Now (#9776, #9980 ) we have a more general solution to define an extra tree SOC, so the original PR is no more necessary.
My opinion is to reduce, in the tree, at minimum possible the files/defines related to a concrete soc. The correct place for this soc (not series) configurations is the board definition files.

**Comment by galak** [comment]
Ok, sounds good!

**Pull request closed without merging**